### DISCUSSION TITLE
A kindly request from the developer

### DISCUSSION THREAD

**Discussion by ferraridamiano** [ferraridamiano/ConverterNOW#43] [Q&A] [answered]
Hi @mezysinc, @Torok42, @FTno, @u2code and thank you all for the translations of the app. Every week someone contribute with new translations and request new features and I want to make the contribution even more frictionless. This is a side project for me and I can work on this just few hours a week. In this period I'm rewriting the app ([see the project](https://github.com/ferraridamiano/ConverterNOW/projects/1)). Part of this work affect also the [translation](https://github.com/ferraridamiano/ConverterNOW/issues/36). I'd like to make them future proof (from .json files to .arb files) and easily translatable. Part of this transition regards repetitive work and if I work on this in these hours I can't fix bugs, add features, etc. I ask you if you want to help me, you don't need any programming skills at all. What you have to do is renaming the *keys* in your language arb file to be the same as the new keys used in the English arb file. Keys are case sensitive. It's a copy/paste work. If the translation in you language is the same as English (e.g. kilobit Is the same in English, Italian, french, etc.) you can omit it. Then pull request the file to *refactoring* branch and not master. I [already setup](https://github.com/ferraridamiano/ConverterNOW/tree/refactoring/lib/l10n) all the files in refactoring branch (don't use master branch!). The files are located in lib/l10n/app_xx.arb. See the [english file](https://github.com/ferraridamiano/ConverterNOW/blob/refactoring/lib/l10n/app_en.arb) as reference (this is already done by me).

This is an experiment for me, I want to make this repo even more yours. If I receive a positive feedback I will involve you even more!

If you have any questions write the below! Thanks for your work and your time! 💚

**Comment by u2code** [comment]
Hello @ferraridamiano I would be glad to help your with the new structure of your project. But there is already an arb-file for the German translation, right?

**Reply by ferraridamiano** [reply]
It is rigth, it is the same file you provided to me. What you have to do is rename the keys of the german translation to be the saame keys as the english arb file.
Thank you for your work!

**Reply by u2code** [reply]
All done! Have a look at the commit.

**Comment by u2code** [comment, marked as answer]
Hi @ferraridamiano 
I updated the strings for the German, French and the Protuguese arb-files. But I have a question.

In the new English arb-file there are strings for bit, byte, etc. and although for different bases deci, mili, etc. This strings were also in the old json-files in the master-branch. In the new arb-files you already created in advance there were no such strings anylonger. There is altough a commit-comment "Removed some unused translation".

My question is are those strings needed in the new-arb files or not?

By the way: I only updated those strings for the German arb-file so far.

**Reply by ferraridamiano** [reply]
Thank you very much @u2code for your work! You saved me a ton of time!
One of the adavantages of the new translation format is that if a translation is missing, it will automatically pick the english word! This is helpful with word such as bit, byte etc. that is the same in almost every language.

Could you help me again if in the future I will need some help in this kind of work? Can I count on you?

**Reply by u2code** [reply]
Sure! Just feel free to ask. I'll do the Norwegian translation in a few moments.

**Reply by ferraridamiano** [reply]
Thank you very much @u2code. Then could you also do italian?
This is not an urgent work, we have not any kind of deadline 😉

**Reply by u2code** [reply]
Sure.

**Reply by u2code** [reply]
Ok it's done. By a fault the Norwegian translation also landed in the italian commit. I deleted it in the italian commit manually. Please pay attention when you are merging!

**Reply by ferraridamiano** [reply]
Super-speedy and excelent work!! Thank you very much for your time and work! 🎉
